engine/baml-schema-wasm/src/runtime_wasm/mod.rs

```diff
@@ -1346,4 +1409,83 @@ impl WasmFunction {
             tracing_project_id: rt.env_vars().get("BOUNDARY_PROJECT_ID").cloned(),
         })
     }
+
+    pub fn orchestration_graph(&self, rt: &WasmRuntime) -> Result<Vec<WasmScope>, JsValue> {
+        let rt: &BamlRuntime = &rt.runtime;
+        let ctx_manager = rt.create_ctx_manager(BamlValue::String("wasm".to_string()));
+        let ctx = ctx_manager.create_ctx_with_default(rt.env_vars().keys().map(|k| k.as_str()));
+        let ir = rt.internal().ir();
+        let walker = ir
+            .find_function(&self.name)
+            .map_err(|e| JsValue::from_str(&format!("{:?}", e)))?;
+        let renderer = PromptRenderer::from_function(&walker, ir, &ctx)
+            .map_err(|e| JsValue::from_str(&format!("{:?}", e)))?;
+        let client_name = renderer.client_name().to_string();
+        log::info!("Client name: {}", client_name);
+
+        let ctx_manager = rt.create_ctx_manager(BamlValue::String("wasm".to_string()));
+        let ctx = ctx_manager.create_ctx_with_default(rt.env_vars().keys().map(|k| k.as_str()));
+
+        let graph = rt
+            .internal()
+            .orchestration_graph(&client_name, &ctx)
+            .map_err(|e| JsValue::from_str(&format!("{:#?}", e)))?;
+
+        // Serialize the scopes to JsValue
+        let mut scopes = Vec::new();
+        for scope in graph {
+            scopes.push(WasmScope::from(scope.scope));
+        }
+        Ok(scopes)
+    }
+}
+trait ToJsValue {
```

Comment:
this is a really non-type-safe way to implement this, but i guess it works? it throws away a lot of type system guarantees we have in rust and relies on our ts code hopefully getting the types correct - that works in a single PR but we're definitely going to mistakes with this in the future, esp. e.g. because we won't have auto-complete on the js analogue of ExecutionScope

Reply:
Will also fix tmrw